File: V семестр/Паралельне програмування/Кочетов/лабы/lab4/lab4/F2.cpp
#include "F2.h"

F2::F2(int N) {
	this->N = N;
}

Matrix* F2::getResult() {
	return result;
}

DWORD WINAPI F2::startThread(void* param) {
	F2* This = (F2*) param;
	return This->run();
}

DWORD F2::run() {
	cout << "Task 2 start\n";
	Matrix *MF = new Matrix(N), *MG = new Matrix(N), *MH = new Matrix(N), *MK = new Matrix(N);
	result = MG->multiply(MF->get_min())->sum(MK->multiply(MF)->multiply(MH->get_max()));
	cout << "Task 2 end\n";
	delete MF;
	delete MG;
	delete MH;
	delete MK;
	return 0;
}

File: V семестр/Паралельне програмування/Бровченко/lab3_OPP/lab3_OPP/lab3_OPP/F2.cs
using System;

namespace lab3
{
    public class F2
    {
    private Matrix result;
    public Matrix getResult()
    {
        return result;
    }

    private int N;

    public F2(int N)
    {
        this.N = N;
    }

    public void run()
    {
            Console.WriteLine("Task 2 start");
            Matrix MF = new Matrix(N), MG = new Matrix(N);
            long k = 3;
            result = MG.multiply(MF.multiply(k));
            Console.WriteLine("Task 2 end");
    }
}

}
